app/src/main/java/com/asfoundation/wallet/promotions/voucher/EVoucherDetailsFragment.kt

```diff
@@ -0,0 +1,97 @@
+package com.asfoundation.wallet.promotions.voucher
+
+import android.content.Intent
+import android.net.Uri
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import android.widget.GridView
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+import com.asf.wallet.R
+import com.jakewharton.rxbinding2.view.RxView
+import io.reactivex.Observable
+import kotlinx.android.synthetic.main.layout_app_bar.*
+
+class EVoucherDetailsFragment : Fragment(), EVoucherDetailsView {
+
+  lateinit var nextButton: Button
+  lateinit var cancelButton: Button
+  lateinit var downloadButton: Button
+  lateinit var gridView: GridView
+  lateinit var diamondsButtonsAdapter: DiamondsButtonsAdapter
+  lateinit var appPackageName: String
+  private lateinit var presenter: EVoucherDetailsPresenter
+
+  override fun onCreateView(
+      inflater: LayoutInflater,
+      container: ViewGroup?,
+      savedInstanceState: Bundle?
+  ): View? {
+    appPackageName = arguments?.getString(PACKAGE_NAME)!!
+
+    return inflater.inflate(R.layout.fragment_e_voucher_details, container, false)
+  }
+
+  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+    super.onViewCreated(view, savedInstanceState)
+
+    presenter = EVoucherDetailsPresenter(
+        this,
+        EVoucherDetailsInteractor(),
+        EVoucherDetailsNavigator()
+    )
+    presenter.present()
+  }
+
```

Comment:
You can already setup the toolbar so the back button (up button I guess) works correctly. You can check PromotionsFragment for an example (setHasOptionsMenu() and overriding onOptionsItemSelected())